platform/darwin/src/MGLTileListOfflineRegion.mm

```diff
@@ -0,0 +1,162 @@
+#import "MGLTileListOfflineRegion.h"
+
+#if !TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+    #import <Cocoa/Cocoa.h>
+#endif
+
+#import "MGLOfflineRegion_Private.h"
+#import "MGLTileListOfflineRegion_Private.h"
+#import "MGLGeometry_Private.h"
+#import "MGLStyle.h"
+#import "MGLLoggingConfiguration_Private.h"
+#import "MGLTileID.h"
+#include <mbgl/tile/tile_id.hpp>
+
+#if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
+#import "MMEConstants.h"
+#endif
+
+@interface MGLTileListOfflineRegion () <MGLOfflineRegion_Private, MGLTileListOfflineRegion_Private>
+
+@end
+
+@implementation MGLTileListOfflineRegion {
+    NSURL *_styleURL;
+}
+
+@synthesize styleURL = _styleURL;
+
+-(NSDictionary *)offlineStartEventAttributes {
+    return @{
+             #if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
+             MMEEventKeyShapeForOfflineRegion: @"tileregion",
+             MMEEventKeyMinZoomLevel: @(self.minimumZoomLevel),
+             MMEEventKeyMaxZoomLevel: @(self.maximumZoomLevel),
+             MMEEventKeyStyleURL: self.styleURL.absoluteString ?: [NSNull null]
+             #endif
+             };
+}
+
++ (BOOL)supportsSecureCoding {
+    return YES;
+}
+
+- (instancetype)init {
+    MGLLogInfo(@"Calling this initializer is not allowed.");
+    [NSException raise:NSGenericException format:
+     @"-[MGLTilePyramidOfflineRegion init] is unavailable. "
+     @"Use -initWithStyleURL:bounds:fromZoomLevel:toZoomLevel: instead."];
+    return nil;
+}
+
+- (instancetype)initWithStyleURL:(NSURL *)styleURL bounds:(MGLCoordinateBounds)bounds fromZoomLevel:(double)minimumZoomLevel toZoomLevel:(double)maximumZoomLevel tileList:(nullable NSArray*)tileList {
+    MGLLogDebug(@"Initializing styleURL: %@ bounds: %@ fromZoomLevel: %f toZoomLevel: %f", styleURL, MGLStringFromCoordinateBounds(bounds), minimumZoomLevel, maximumZoomLevel);
+    if (self = [super init]) {
+        if (!styleURL) {
+            styleURL = [MGLStyle streetsStyleURLWithVersion:MGLStyleDefaultVersion];
+        }
+
+        if (!styleURL.scheme) {
```

Comment:
does streetsStyleURLWithVersion possible return a scheme-less URL that we need to check, too? If not, this block might be clearer as:

if (!styleURL) {
 ...
} else if (!styleURL.scheme) {
 ...
}